gc/base/CardTable.hpp

```diff
@@ -207,6 +207,23 @@ class MM_CardTable : public MM_BaseVirtual
 	 */
 	bool canMemoryBeReleased(MM_EnvironmentBase *env, void *low, void *high);
 	
+	/**
+	 * alignment the address with CARD_SIZE
+	 * @param toFloor = true, align to floor; =false, align to ceiling
+	 * @param alignmentMultiple The multiple to which the _allocatePointer must be aligned (must be a power of 2)
+	 */
+	static void* alignWithCard(void *address, bool toFloor, uintptr_t alignmentMultiple)
```

Comment:
Let's specialize methods for to-floor and to-ceiling to avoid toFloor runtime check.
Keep the methods in MM_Math. We already have similar like roundToFloorU64, but just call it roundToFloorCard (no need for argument then).
As you are already aware we don't want to used generic roundToFloor, since we know that our alignment is a power of 2.

Reply:
Well, you can keep this API if you want, but just call Math.